tests/Http/Three/SignupTest.php

```diff
@@ -105,4 +105,39 @@ public function testSignupWithIncludedPosts()
             ],
         ]);
     }
+
+    /**
+     * Test that a signup gets deleted when hitting the DELETE endpoint.
+     *
+     * @return void
+     */
+    public function testDeletingASignup()
+    {
+        $signup = factory(Signup::class)->create();
+
+        // Mock time of when the signup is soft deleted.
+        $this->mockTime('8/3/2017 14:00:00');
+
+        $response = $this->withRogueApiKey()->delete('api/v3/signups/' . $signup->id);
+
+        $this->assertResponseStatus(200);
+
+        // Make sure that the signup's deleted_at gets persisted in the database.
+        $this->assertEquals($signup->fresh()->deleted_at->toTimeString(), '14:00:00');
+
+    }
+
+    /**
+     * Test that non-authenticated user's/apps can't delete signups.
+     *
+     * @return void
+     */
+    public function testUnauthenticatedUserDeletingASignup()
+    {
+        $signup = factory(Signup::class)->create();
+
+        $response = $this->delete('api/v3/signups/' . $signup->id);
+
+        $response->assertResponseStatus(401);
```

Comment:
When I test this in postman, it throws a `401` response if the `X-DS-Rogue-API-Key` is not present in the header. However, here, it throws a `302` error code and I'm not sure why. I'm wondering if I need to add a function to `BrowserKitTestCase` of `withoutRogueApiKey` similar to `withRogueApiKey` but if it doesn't have this function, it shouldn't have that header and should throw the `401` anyway?

Reply:
I think that this is throwing a `302` because it is trying to re-route the page to the login page when the test runs - but really would like this response like in Postman:
```
{
    "error": {
        "code": 401,
        "message": "Unauthorized."
    },
    "debug": {
        "file": "/home/vagrant/DoSomething.org/rogue/app/Exceptions/Handler.php",
        "line": 85
    }
}
```

Reply:
I think the problem is that the `delete` test method doesn't send along `Accept: application/json` so we get the "web" response (whereas you're probably including that when you test in Postman). Let's use `deleteJson` instead to make sure we get that header:

```php
$response = $this->deleteJson('api/v3/signups/' . $signup->id);
```

Reply:
ooo yes, thank you so much and for the explanation too, @DFurnes ! That worked!